mkvparser/mkvparser.h

```diff
@@ -1054,11 +1054,17 @@ class Segment {
 
   Segment(const Segment&);
   Segment& operator=(const Segment&);
+ public:
+  enum Type
```

Comment:
Did you consider instead of adding this enum, expose methods for caller to remove unused clusters? 

Unfortunately those are seemingly missing in existing code as well, and require update (if I am not missing anything) but they are not as interfering as the enum,.

Out demultiplexer will just check cluster count and remove those we no longer need. We can even push this back to opus.

Reply:
I could add that.

Reply:
I made some notes on what we'd need to do to support unloading of Clusters.
```C++
Segment::GetBlock - binary search to find a particular CuePoint and TrackPosition.
Segment::FindOrPreloadCluster - binary search to find a particular cluster (duplicate code).
Segment::GetFirst/Last - assert non-null
Segment::GetNext - same binary search code.
Segment::DoParseNext - same binary search code duplicated again.
Track::Seek - same binary search code duplicated again.
VideoTrack::Seek - same binary search code duplicated again.
```

Reply:
Actually on second thought, after considering the other things on my TODO list I'm gonna leave this as a TODO for now.